acvm/src/lib.rs

```diff
@@ -92,7 +92,18 @@ fn first_missing_assignment(
     })
 }
 
-pub trait Backend: SmartContract + ProofSystemCompiler + PartialWitnessGenerator + Default {}
+pub trait Backend:
+    SmartContract + ProofSystemCompiler + PartialWitnessGenerator + ReferenceString + Default
+{
+}
+
+pub trait ReferenceString {
+    type ReferenceString; // TODO: Figure out serde trait
+
+    // TODO: Figure out Err type
+    // TODO: Figure out async trait
+    fn get_reference_string(&self, circuit: &Circuit) -> Result<Self::ReferenceString, String>;
```

Comment:
proposal: `can_compute_circuit(&self, reference_string: &Self::ReferenceString, circuit: &Circuit)`

Reply:
Though, the above proposed API doesn't account for appending more `g1_data` to the data we already have; instead opting to download all of the data to fulfill a larger circuit.

Reply:
It sounds like we could adapt `get_reference_string` or include a new method to account for appending data rather than re-downloading data we already have?

Reply:
I've changed the interface to be `generate_common_reference_string` and `update_common_reference_string` to account for this. It is worth noting that Nargo CLI will need to write data in every scenario with this API but it will lighten network loads, so I think it is a fine trade-off.